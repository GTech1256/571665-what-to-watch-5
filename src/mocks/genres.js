import {FIRST_FILTER_NAME} from "../const";

export default [FIRST_FILTER_NAME, `Comedy`, `Fantasy`];
