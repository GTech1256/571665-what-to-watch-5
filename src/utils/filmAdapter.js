/* eslint-disable camelcase */

export const adaptFilmToServer = ({
  id,
  name,
  posterImage,
  previewImage,
  backgroundImage,
  backgroundColor,
  videoLink,
  previewVideoLink,
  description,
  rating,
  scoresCount,
  director,
  starring,
  runTime,
  genre,
  released,
  isFavorite
}) => ({
  id,
  name,
  poster_image: posterImage,
  preview_image: previewImage,
  background_image: backgroundImage,
  background_color: backgroundColor,
  description,
  rating,
  scores_count: scoresCount,
  director,
  starring,
  run_time: runTime,
  genre,
  released,
  is_favorite: isFavorite,
  video_link: videoLink,
  preview_video_link: previewVideoLink
});


export const adaptFilmFromServer = ({
  id,
  name,
  poster_image,
  preview_image,
  background_image,
  background_color,
  description,
  rating,
  scores_count,
  director,
  starring,
  run_time,
  genre,
  released,
  is_favorite,
  video_link,
  preview_video_link
}) => ({
  id,
  name,
  posterImage: poster_image,
  previewImage: preview_image,
  backgroundImage: background_image,
  backgroundColor: background_color,
  videoLink: video_link,
  previewVideoLink: preview_video_link,
  description,
  rating,
  scoresCount: scores_count,
  director,
  starring,
  runTime: run_time,
  genre,
  released,
  isFavorite: is_favorite
});

