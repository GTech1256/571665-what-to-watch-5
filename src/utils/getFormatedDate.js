const DATE_TIME_FORMAT_LOCALE = `en-US`;
const DATE_TIME_FORMAT_OPTION = {
  month: `long`,
  day: `numeric`,
  year: `numeric`
};

export const getFormatedDate = (value) => new Intl.DateTimeFormat(
    DATE_TIME_FORMAT_LOCALE,
    DATE_TIME_FORMAT_OPTION
)
  .format(
      new Date(value)
  );
