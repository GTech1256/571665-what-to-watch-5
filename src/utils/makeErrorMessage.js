export const makeErrorMessage = (error) => `${error.message}${error.response ? `: ${error.response.data.error}` : ``}`;
