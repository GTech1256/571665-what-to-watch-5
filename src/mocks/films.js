import {extend} from "../utils/extend";
import film from "./film";

export default [film, film, film, film].map((filmMock, i) => extend(filmMock, {id: `${i}`}));
