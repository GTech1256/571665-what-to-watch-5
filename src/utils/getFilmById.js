import films from "../mocks/films";

export const getFilmById = (filmId) => films.find(({id}) => filmId === `${id}`);
