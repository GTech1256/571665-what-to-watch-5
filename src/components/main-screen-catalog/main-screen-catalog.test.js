import React from "react";
import {MemoryRouter} from "react-router-dom";
import renderer from "react-test-renderer";
import films from "../../mocks/films";
import genres from "../../mocks/genres";
import {noop} from "../../utils/noop";
import MainScreenCatalog from "./main-screen-catalog";

describe(`MainScreenCatalog is rendered correctly`, () => {
  it(`with ShowMoreBtn`, () => {
    const tree = renderer
    .create(
        <MemoryRouter>
          <MainScreenCatalog
            films={films}
            activeGenre={genres[0]}
            genres={genres}
            isShowShowMoreBtn
            onGenreClick={noop}
            onShowMoreBtnClick={noop}
          />
        </MemoryRouter>,
        {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`without ShowMoreBtn`, () => {
    const tree = renderer
    .create(
        <MemoryRouter>
          <MainScreenCatalog
            films={films}
            activeGenre={genres[0]}
            genres={genres}
            onGenreClick={noop}
            onShowMoreBtnClick={noop}
          />
        </MemoryRouter>,
        {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
