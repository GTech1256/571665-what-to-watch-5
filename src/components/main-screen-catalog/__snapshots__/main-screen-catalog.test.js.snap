// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreenCatalog is rendered correctly with ShowMoreBtn 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item catalog__genres-item--active"
    >
      <a
        className="catalog__genres-link"
        onClick={[Function]}
      >
        All Genres
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        onClick={[Function]}
      >
        Comedy
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        onClick={[Function]}
      >
        Fantasy
      </a>
    </li>
  </ul>
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
    >
      <a
        href="/films/0"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            height="175"
            poster="img/the-grand-budapest-hotel-poster.jpg"
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            width="280"
          />
        </div>
      </a>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/0"
          onClick={[Function]}
        >
          The Grand Budapest Hotel
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <a
        href="/films/1"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            height="175"
            poster="img/the-grand-budapest-hotel-poster.jpg"
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            width="280"
          />
        </div>
      </a>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/1"
          onClick={[Function]}
        >
          The Grand Budapest Hotel
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <a
        href="/films/2"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            height="175"
            poster="img/the-grand-budapest-hotel-poster.jpg"
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            width="280"
          />
        </div>
      </a>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/2"
          onClick={[Function]}
        >
          The Grand Budapest Hotel
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <a
        href="/films/3"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            height="175"
            poster="img/the-grand-budapest-hotel-poster.jpg"
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            width="280"
          />
        </div>
      </a>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/3"
          onClick={[Function]}
        >
          The Grand Budapest Hotel
        </a>
      </h3>
    </article>
  </div>
  <div
    className="catalog__more"
  >
    <button
      className="catalog__button"
      onClick={[Function]}
      type="button"
    >
      Show more
    </button>
  </div>
</section>
`;

exports[`MainScreenCatalog is rendered correctly without ShowMoreBtn 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item catalog__genres-item--active"
    >
      <a
        className="catalog__genres-link"
        onClick={[Function]}
      >
        All Genres
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        onClick={[Function]}
      >
        Comedy
      </a>
    </li>
    <li
      className="catalog__genres-item "
    >
      <a
        className="catalog__genres-link"
        onClick={[Function]}
      >
        Fantasy
      </a>
    </li>
  </ul>
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
    >
      <a
        href="/films/0"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            height="175"
            poster="img/the-grand-budapest-hotel-poster.jpg"
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            width="280"
          />
        </div>
      </a>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/0"
          onClick={[Function]}
        >
          The Grand Budapest Hotel
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <a
        href="/films/1"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            height="175"
            poster="img/the-grand-budapest-hotel-poster.jpg"
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            width="280"
          />
        </div>
      </a>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/1"
          onClick={[Function]}
        >
          The Grand Budapest Hotel
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <a
        href="/films/2"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            height="175"
            poster="img/the-grand-budapest-hotel-poster.jpg"
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            width="280"
          />
        </div>
      </a>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/2"
          onClick={[Function]}
        >
          The Grand Budapest Hotel
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
    >
      <a
        href="/films/3"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            height="175"
            poster="img/the-grand-budapest-hotel-poster.jpg"
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            width="280"
          />
        </div>
      </a>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/3"
          onClick={[Function]}
        >
          The Grand Budapest Hotel
        </a>
      </h3>
    </article>
  </div>
</section>
`;
